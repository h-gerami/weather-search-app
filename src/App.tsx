import * as React from 'react';
import {SafeAreaView, StatusBar, StyleSheet} from 'react-native';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';
import {NavigationContainer} from '@react-navigation/native';
import {store, persistor} from './Redux/Store';
import Tabs from './Navigation/Tabs';

function App() {
  return (
    <Provider store={store}>
      <StatusBar backgroundColor="#fff" barStyle="dark-content" />
      <PersistGate loading={null} persistor={persistor}>
        <NavigationContainer>
          <SafeAreaView style={styles.container}>
            <Tabs />
          </SafeAreaView>
        </NavigationContainer>
      </PersistGate>
    </Provider>
  );
}

export default App;
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
