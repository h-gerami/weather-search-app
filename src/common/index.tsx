export * from './Button/ProfileButton';
export * from './Button/NotificationButton';
export * from './Button/Button';
