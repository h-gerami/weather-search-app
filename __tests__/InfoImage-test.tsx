import React from 'react';
import renderer from 'react-test-renderer';
import {InfoImage} from '../src/common';

test('renders correctly', () => {
  const tree = renderer.create(<InfoImage isPortrait={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
