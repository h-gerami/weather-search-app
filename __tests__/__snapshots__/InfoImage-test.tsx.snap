// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 20,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/cload.png",
      }
    }
    style={
      Object {
        "height": 100,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#1C3041",
        "fontFamily": "SofiaProMedium",
        "fontSize": 15,
        "lineHeight": 20,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Please Enter your City Name and then Click the Search Button
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/sun.png",
      }
    }
    style={
      Object {
        "height": 100,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
</View>
`;
